src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -679,3 +709,113 @@ def forward(
             cache_params=mamba_outputs.cache_params,
             hidden_states=mamba_outputs.hidden_states,
         )
+
+
+class MambaClassificationHead(nn.Module):
+    """Head for sentence-level classification tasks."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.activation = ACT2FN[config.hidden_act]
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+        self.out_proj = nn.Linear(config.hidden_size, config.num_labels)
+
+        self.config = config
+
+    def forward(self, features, **kwargs):
+        x = features[:, 0, :]  # take <s> token (equiv. to [CLS])
+        x = self.dropout(x)
+        x = self.dense(x)
+        x = self.activation(x)
+        x = self.dropout(x)
+        x = self.out_proj(x)
+        return x
+
+
+@add_start_docstrings(
+    """Mamba Model backbone with a sequence classification/regression head on top (a linear layer on top of
+    the pooled output) e.g. for GLUE tasks.""",
+    MAMBA_START_DOCSTRING,
+)
+class MambaForSequenceClassification(MambaPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.backbone = MambaModel(config)
+        self.classifier = MambaClassificationHead(config)
+
+        for param in self.base_model.parameters():
+            param.requires_grad = False
```

Comment:
I'm not sure whether we actually want to freeze the params for the base model here, but I found there's [a test specific to sequence classification that expects all the unfrozen params to be initialized in the range `[0.0, 1.0]` ](https://app.circleci.com/pipelines/github/huggingface/transformers/87066/workflows/ee804f42-0535-452c-8ee6-38df520712e3/jobs/1130735/parallel-runs/0/steps/0-115) and the initialized values for the mixer don't satisfy that assertion.

So... I froze them and made sure [the classification head params were initialized to satisfiy the test](https://github.com/huggingface/transformers/pull/29552/files#diff-e1d4758c08973fdac2c23a8a3710872d943ce8509035205da4a681bc4dcaf1c3R365-R368).  It makes intuitive sense to me to freeze them in the case of transfer learning for this task and I did confirm that running LoRA PEFT with `target_modules=["x_proj", "embeddings", "in_proj", "out_proj"]` and `task_type=TaskType.SEQ_CLS` does unfreeze the target modules so it appears to work fine, but not sure if we want to force them to be frozen by default. 

Anyway, happy to adjust if there's a better practice to follow here.